### PR TITLE
dev-util/source-highlight: fix lto, drop ctags dependency to optfeature

### PR DESCRIPTION
**Pull request by eli-schwartz**

### CODE REVIEW

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild:4]: 8 while we're here

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	# That's not how we want it
+	rm -rf "${ED}"/usr/share/{aclocal,doc} || die
+
+	# package provides .pc file
+	find "${D}" -name '*.la' -delete || die
+
+	dobashcomp completion/source-highlight
+}
+
+src_test() {
```
Comment (thesamesam) [inline dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild:58]: Hoist this above src_install to reflect phase execution order

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit bash-completion-r1 flag-o-matic libtool optfeature
+
+DESCRIPTION="Generate highlighted source code as an (x)html document"
+HOMEPAGE="https://www.gnu.org/software/src-highlite/source-highlight.html"
+SRC_URI="mirror://gnu/src-highlite/${P}.tar.gz"
+LICENSE="GPL-3"
```
Comment (thesamesam) [inline dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild:11]: Newline above LICENSE

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+BDEPEND="test? ( dev-util/ctags )"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-3.1.9-test-clang-p1.patch
+	"${FILESDIR}"/${PN}-3.1.9-test-clang-p2.patch
+)
+
+src_prepare() {
+	default
+	elibtoolize
```
Comment (thesamesam) [inline dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild:28]: Maybe add a comment above for why?

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild (hunk 2 above)
Comment (eli-schwartz) [reply dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild]: Sure, while we're here and all.

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild (hunk 3 above)
Comment (eli-schwartz) [reply dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild:12]: Done.

File: dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild (hunk 4 above)
Comment (eli-schwartz) [reply dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild:32]: Done.

**Comment by eli-schwartz** [comment]
Range-diff:
```
1:  f478f904a94a = 1:  0be5c74bdfaf dev-util/source-highlight: remove unneeded runtime dependency on ctags
2:  d60e5233c5e4 ! 2:  ebf29e80a765 dev-util/source-highlight: apply elibtoolize to old distributed libtool
    @@ dev-util/source-highlight/source-highlight-3.1.9-r2.ebuild: PATCHES=(
      
     +src_prepare() {
     +	default
    ++
    ++	# Although all unpatched libtools are probably broken, this one ignores LTO
    ++	# warning flags.
     +	elibtoolize
     +}
     +
3:  d2232b8d7b4a = 3:  9e0d99094f10 dev-util/source-highlight: disallow building with LTO
-:  ------------ > 4:  38d504eab499 dev-util/source-highlight: fix some old formatting quirks
-:  ------------ > 5:  efabc11d7b34 dev-util/source-highlight: update EAPI 7 -> 8
```

**Pull request closed without merging**